tests_e2e/orchestrator/lib/agent_test_suite_combinator.py

```diff
@@ -216,27 +239,35 @@ def create_environment_list(self) -> List[Dict[str, Any]]:
                     else:
                         shared_environments[key] = {
                             "c_marketplace_image": marketplace_image,
+                            "c_cloud": self.runbook.cloud,
                             "c_location": location,
                             "c_vm_size": vm_size,
                             "c_vhd": vhd,
                             "c_test_suites": [suite_info],
-                            "c_env_name": key
+                            "c_env_name": key,
+                            "c_marketplace_image_information_location": self._MARKETPLACE_IMAGE_INFORMATION_LOCATIONS[self.runbook.cloud],
+                            "c_shared_resource_group_location": self._SHARED_RESOURCE_GROUP_LOCATIONS[self.runbook.cloud]
                         }
 
         environment_list.extend(shared_environments.values())
 
+        if len(environment_list) == 0:
+            raise Exception("No VM images were found to execute the test suites.")
+
         log: logging.Logger = logging.getLogger("lisa")
-        log.info("******** Environments *****")
-        for e in environment_list:
-            log.info(
-                "{ c_marketplace_image: '%s', c_location: '%s', c_vm_size: '%s', c_vhd: '%s', c_test_suites: '%s', c_env_name: '%s' }",
-                e['c_marketplace_image'], e['c_location'], e['c_vm_size'], e['c_vhd'], [s.name for s in e['c_test_suites']], e['c_env_name'])
+        log.info("")
```

Comment:
Now I am logging the entire list, instead of explicitly specifying each item in the list.